src/main/java/org/jenkinsci/plugins/blueoceandisplayurl/EnvironmentContributorImpl.java

```diff
@@ -0,0 +1,28 @@
+package org.jenkinsci.plugins.blueoceandisplayurl;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.EnvironmentContributor;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLContext;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+@Extension
+public class EnvironmentContributorImpl extends EnvironmentContributor {
+    @Override
+    public void buildEnvironmentFor(@Nonnull Run r, @Nonnull EnvVars envs, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+        DisplayURLContext ctx = DisplayURLContext.open();
+        try {
+            ctx.run(r);
+            ctx.plugin(null); // environment contributor "comes from" core
+            BlueOceanDisplayURLImpl urlProvider = new BlueOceanDisplayURLImpl();
```

Comment:
it seems to me from this code, that the display url context is not actually used, is it a global somewhere?

Reply:
I actually used the same approach as done in https://github.com/jenkinsci/display-url-api-plugin/blob/baa22c4f2e54f580a1a0071fc16f7064e6c5cf07/src/main/java/org/jenkinsci/plugins/displayurlapi/EnvironmentContributorImpl.java#L16-L27

Not sure whether that's what you meant

Reply:
yea.. thats sorta what I mean, i'll have to poke and see what open/run/plugin is doing.